c_glib/arrow-glib/basic-data-type.cpp

```diff
@@ -97,6 +97,15 @@ G_BEGIN_DECLS
  * #GArrowTime64DataType is a class for the number of microseconds or
  * nanoseconds since midnight in the 64-bit signed integer data type.
  *
+ * #GArrowIntervalDataType is an abstract class for interval related data type
+ * such as #GArrowIntervalMonthsDataType.
+ *
+ * #GArrowIntervalMonthsDataType is a class for the month intarval data type.
```

Comment:
We should use `GArrowXXXIntervalDataType` for consistency.
For example, we already have `GArrowListDataType` and `GArrowLargeListDataType`.

For `arrow::IntervalType::MONTHS`, let's use `MONTH` not `MONTHS` in GLib for consistency. Other interval types don't have `S`.

We can discuss C++ API (`arrow::IntervalType::MONTHS` or `...::MONTH`, which is better?) later with C++ developers on Jira or mailing list.